homeassistant/components/websocket_api/commands.py

```diff
@@ -127,54 +131,84 @@ def handle_unsubscribe_events(hass, connection, msg):
 @decorators.async_response
 async def handle_call_service(hass, connection, msg):
     """Handle call service command."""
-    blocking = True
-    if msg["domain"] == HASS_DOMAIN and msg["service"] in ["restart", "stop"]:
-        blocking = False
-
     # We do not support templates.
     target = msg.get("target")
     if template.is_complex(target):
         raise vol.Invalid("Templates are not supported here")
 
+    def _prepare_service_data():
+        return msg["domain"], msg["service"], msg.get("service_data"), target
+
+    await _handle_call_service(hass, connection, msg["id"], _prepare_service_data)
+
+
+@decorators.websocket_command(
+    {
+        vol.Required("type"): "call_service_action",
```

Comment:
Would it be better named something like `execute_script` ?

Reply:
Yes, if we change it from allowing execution of a single script `service` action to execution of a script. That was not the intention of the PR though.